com.sap.cloud.lm.sl.cf.process/src/main/java/com/sap/cloud/lm/sl/cf/process/steps/UpdateServicePlanStep.java

```diff
@@ -0,0 +1,67 @@
+package com.sap.cloud.lm.sl.cf.process.steps;
+
+import java.text.MessageFormat;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Named;
+
+import org.cloudfoundry.client.lib.CloudControllerClient;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Component;
+
+import com.sap.cloud.lm.sl.cf.client.lib.domain.CloudServiceExtended;
+import com.sap.cloud.lm.sl.cf.core.cf.clients.ServiceInstanceGetter;
+import com.sap.cloud.lm.sl.cf.core.cf.clients.ServiceUpdater;
+import com.sap.cloud.lm.sl.cf.core.cf.services.ServiceOperationType;
+import com.sap.cloud.lm.sl.cf.process.message.Messages;
+import com.sap.cloud.lm.sl.common.util.JsonUtil;
+
+@Component("updateServicePlanStep")
+public class UpdateServicePlanStep extends AsyncFlowableStep {
+    
+    @Inject
+    @Named("serviceUpdater")
+    protected ServiceUpdater serviceUpdater; 
+
+    @Inject
+    private ServiceInstanceGetter serviceInstanceGetter;
+
+    @Override
+    protected StepPhase executeAsyncStep(ExecutionWrapper execution) throws Exception {
+        CloudServiceExtended serviceToUpdate = StepsUtil.getServiceToProcess(execution.getContext());
+        ResponseEntity<String> updateServicePlanResponse = updateServicePlan(execution.getControllerClient(), serviceToUpdate);
+        
+        if(HttpStatus.ACCEPTED.equals(updateServicePlanResponse.getStatusCode())) {
+            Map<String, ServiceOperationType> serviceOperation = new HashMap<>();
+            serviceOperation.put(serviceToUpdate.getName(), ServiceOperationType.UPDATE);
+            
+            execution.getStepLogger()
+            .debug(Messages.TRIGGERED_SERVICE_OPERATIONS, JsonUtil.toJson(serviceOperation, true));
+            StepsUtil.setTriggeredServiceOperations(execution.getContext(), serviceOperation);
+            
+            return StepPhase.POLL;
+        }
+        return StepPhase.DONE;
+    }
+
+    private ResponseEntity<String> updateServicePlan(CloudControllerClient client, CloudServiceExtended service) {
+        getStepLogger()
+            .debug(MessageFormat.format("Updating service plan of a service {0} with new plan: {1}", service.getName(), service.getPlan()));
+        if (service.shouldIgnoreUpdateErrors()) {
```

Comment:
No no no, not like this :) remove the else

Reply:
Why ?

Reply:
Done.